src/kernel/src/arch/amd64/tsc.rs

```diff
@@ -0,0 +1,127 @@
+use crate::time::{ClockHardware, Ticks};
+
+use twizzler_abi::syscall::{ClockInfo, FemtoSeconds};
+
+// 1 ms = 1000000 ns
+// 200 milliseconds
+const SLEEP_TIME: u64 = 200 * 1_000_000; 
+
+// resolution expressed as a unit of time
+pub struct Tsc {
+    resolution: FemtoSeconds
+}
+
+impl Tsc {
+    pub fn new() -> Self {
+        // calculate the frequency at which the TSC is running
+        // in other words the resolution at which ticks occur
+        let f = Tsc::get_tsc_frequency();
```

Comment:
Please do not use single character variables.  If f is frequency at least give us "freq" to read.